### ISSUE TITLE
issues about nested types

### ISSUE THREAD

**Issue by virusdefender** [teal-language/tl#121]
`someds.d.tl`

```lua
local someds = record
    event_t = record
    end
    callback_t = functiontype(event_t)
    subscribe: function(callback: callback_t)
end

return someds
```

`main.tl`

```lua
local someds = require("someds")

function main()
    local b:someds.callback_t = function(event: someds.event_t)
    end
    someds.subscribe(b)
end
```

`tl check` will show `argument 1: argument 1: someds.event_t is not a event_t`

**Comment by hishamhm** [comment]
Thank you for the report! I'll look into this.

**Comment by hishamhm** [comment]
This was also fixed by the same fix for #122 !

**Comment by virusdefender** [comment]
thank you,  I'm sure it works now 🚀🚀🚀🚀🚀
